pkg/services/control/server/status.go

```diff
@@ -0,0 +1,95 @@
+package control
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-node/pkg/services/control"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *Server) ObjectStatus(_ context.Context, request *control.ObjectStatusRequest) (*control.ObjectStatusResponse, error) {
+	err := s.isValidRequest(request)
+	if err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	// check availability
+	err = s.ready()
+	if err != nil {
+		return nil, err
+	}
+
+	var addr oid.Address
+	err = addr.DecodeString(string(request.Body.ObjectAddress))
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("parsing object address: %s", err.Error()))
+	}
+
+	st, err := s.storage.ObjectStatus(addr)
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("storage engine error: %s", err.Error()))
+	}
+
+	resp := &control.ObjectStatusResponse{
+		Body: &control.ObjectStatusResponse_Body{},
+	}
+
+	for _, sh := range st.Shards {
+		respSh := new(control.ObjectStatusResponse_Body_Shard)
+		respSh.ShardId = sh.ID
+
+		if len(sh.Shard.Metabase.State) == 0 {
+			// can be reconsidered since it is possible to get
+			// resynchronized state when meta knows nothing about
+			// stored objects in blob; however, it is a control
+			// service, not a debug util
+			continue
```

Comment:
IIUC ur comment: if it is in write cache and/or blob storage, some info should be returned anyway. We can fallback this case to `MISSING` status for metabase

Reply:
> if it is in write cache and/or blob storage, some info should be returned anyway

that is why i left comment: i do not think we should search for a object if a metabase in this shard does not know it. this is not `neofs-lens`, it is a control service

Reply:
> // can be reconsidered

what do u mean then?

Reply:
i meant somebody can disagree and we may change it in the future. if too many devs disagree now, may change it now

Reply:
let bs/wc aint in sync with mb - what output will be currently shown?

Reply:
> let bs/wc aint in sync with mb

if bs/wc _has_ info, but metabase does not know about an object, it will be `<empty response>` currently

> And what GET would respond

it really depends on the degree of resynchronization, it meta know absolutely nothing about an object, than it is 404: https://github.com/nspcc-dev/neofs-node/blob/73e84148f8564c5ae71e8b537b6cf0014a96dd47/pkg/local_object_storage/shard/get.go#L146-L148